code/modules/mob/living/simple_animal/hostile/ooze.dm

```diff
@@ -0,0 +1,482 @@
+/mob/living/simple_animal/hostile/ooze/gelatinous/Initialize()
+	. = ..()
+	boost = new
+	boost.Grant(src)
+	consume = new
+	consume.Grant(src)
+
+///If this mob gets resisted by something, its trying to escape consumption.
+/mob/living/simple_animal/hostile/ooze/gelatinous/resist_act(mob/living/user)
+	. = ..()
+	if(!do_after(user, 60)) //6 second struggle
+		return FALSE
+	consume.stop_consuming()
+
+/mob/living/simple_animal/hostile/ooze/gelatinous/add_cell_sample()
+	AddElement(/datum/element/swabable, CELL_LINE_TABLE_GELATINOUS, CELL_VIRUS_TABLE_GENERIC_MOB, 1, 5)
+
+///This ability lets the gelatinious ooze speed up for a little bit
+/datum/action/cooldown/metabolicboost
+	name = "Metabolic boost"
+	desc = "Gain a temporary speed boost. Costs 10 nutrition and slowly raises your temperature"
+	background_icon_state = "bg_hive"
+	icon_icon = 'icons/mob/actions/actions_slime.dmi'
+	button_icon_state = "metabolic_boost"
+	check_flags = AB_CHECK_CONSCIOUS|AB_CHECK_STUN
+	cooldown_time = 240
+	var/nutrition_cost = 10
+	var/active = FALSE
+
+
+///Mob needs to have enough nutrition
+/datum/action/cooldown/metabolicboost/IsAvailable()
+	. = ..()
+	var/mob/living/simple_animal/hostile/ooze/ooze = owner
+	if(. && ooze.ooze_nutrition >= nutrition_cost && !active)
+		return TRUE
+
+///Give the mob a speed boost, heat it up every second, and end the ability in 6 seconds
+/datum/action/cooldown/metabolicboost/Trigger()
+	. = ..()
+	if(!.)
+		return
+	var/mob/living/simple_animal/hostile/ooze/ooze = owner
+	ooze.add_movespeed_modifier(/datum/movespeed_modifier/metabolicboost)
+	var/timerid = addtimer(CALLBACK(src, .proc/HeatUp), 1 SECONDS, TIMER_STOPPABLE | TIMER_LOOP) //Heat up every second
+	addtimer(CALLBACK(src, .proc/FinishSpeedup, timerid), 6 SECONDS)
+	to_chat(ooze, "<span class='notice>You start feel a lot quicker.</span>")
+	ooze.adjust_ooze_nutrition(-10)
+
+///Heat up the mob a little
+/datum/action/cooldown/metabolicboost/proc/HeatUp()
+	var/mob/living/simple_animal/hostile/ooze/ooze = owner
+	ooze.adjust_bodytemperature(50)
+
+///Remove the speed modifier and delete the timer for heating up
+/datum/action/cooldown/metabolicboost/proc/FinishSpeedup(timerid)
+	var/mob/living/simple_animal/hostile/ooze/ooze = owner
+	ooze.remove_movespeed_modifier(/datum/movespeed_modifier/metabolicboost)
+	to_chat(ooze, "<span class='notice>You start slowing down again.</span>")
+	deltimer(timerid)
+
+
+///This action lets you consume the mob you're currently pulling. I'M GONNA CONSUUUUUME (this is considered one of the funny memes in the 2019-2020 era)
+/datum/action/consume
+	name = "Consume"
+	desc = "Consume a mob that you are dragging to gain nutrition from them"
+	background_icon_state = "bg_hive"
+	icon_icon = 'icons/mob/actions/actions_slime.dmi'
+	button_icon_state = "consume"
+	check_flags = AB_CHECK_CONSCIOUS|AB_CHECK_STUN
+	///The mob thats being consumed by this creature
+	var/mob/living/vored_mob
```

Comment:
Noooo!!! Help meee!!!
Also, clean the reference on `Destroy()`

Reply:
ok